src/main/java/com/laytonsmith/core/functions/PlayerManagement.java

```diff
@@ -4897,4 +4898,198 @@ public CHVersion since() {
 			return CHVersion.V3_3_1;
 		}
 	}
+	
+	@api
+    public static class set_pgliding extends AbstractFunction {
+		public String getName() {
+            return "set_pgliding";
+        }
+		
+		public Integer[] numArgs() {
+            return new Integer[] {1, 2};
+        }
+      
+        public String docs() {
+            return "void {[Player], boolean} Sets the player to glide";
+        }
+		
+        public Class<? extends CREThrowable>[] thrown() {
+            return new Class[] {
+                    CRECastException.class
+            };
+        }
+      
+        public boolean isRestricted() {
+            return false;
+        }
+      
+        public Boolean runAsync() {
+            return false;
+        }
+      
+        public Construct exec(Target t, Environment environment, Construct... args) throws ConfigRuntimeException {
+            MCPlayer p;
+            boolean glide;
+            
+            if(args.length == 1) {
+                p = ((CommandHelperEnvironment)environment.getEnv(CommandHelperEnvironment.class)).GetPlayer();
+                glide = Static.getBoolean(args[0]);
+            } else {
+                p = Static.GetPlayer(args[0].val(), t);
+                glide = Static.getBoolean(args[1]);
+            }
+            
+            if(p.getEquipment().getChestplate().getTypeId() != 443 || p.isOnGround()) {
```

Comment:
This generates an NPE if equipment or chestplate is null. Also you should absolutely not be using numeric item ids. That check should probably not even be there since other items could be added that would allow gliding.